src/prebid.js

```diff
@@ -670,10 +684,45 @@ pbjs.addAdUnits = function(adUnitArr) {
 	}
 };
 
+/**
+ * @param {String} event the name of the event
+ * @param {Function} handler a callback to set on event
+ * @param {String} id an identifier in the context of the event
+ *
+ * This API call allows you to register a callback to handle a Prebid.js event.
+ * An optional `id` parameter provides more finely-grained event callback registration.
+ * This makes it possible to register callback events for a specific item in the
+ * event context. For example, `bidWon` events will accept an `id` for ad unit code.
+ * `bidWon` callbacks registered with an ad unit code id will be called when a bid
+ * for that ad unit code wins the auction. Without an `id` this method registers the
+ * callback for every `bidWon` event.
+ *
+ * Currently `bidWon` is the only event that accepts an `id` parameter.
+ */
+pbjs.onEvent = function(event, handler, id) {
+	if(id && !eventValidators[event].call(null, id)){
+		return;
```

Comment:
An error is logged in the event validator function, in this case `checkDefinedPlacement()`, as the specific message may vary. I'll add a general message that `onEvent` failed as well as a check for `typeof handler === 'function'`.